src/main/java/org/fusesource/jansi/io/HtmlAnsiOutputStream.java

```diff
@@ -0,0 +1,176 @@
+/*
+ * Copyright (C) 2009-2017 the original author(s).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fusesource.jansi.io;
+
+import org.fusesource.jansi.AnsiConsole;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * A optimized/edited {@link HtmlAnsiOutputStream} for ansi 2.x and above. <br>
+ * @author <a href="https://github.com/Osiris-Team">Osiris Team</a>
+ * @author <a href="http://code.dblock.org">Daniel Doubrovkine</a>
+ */
+public class HtmlAnsiOutputStream extends AnsiOutputStream {
+
+    @Override
+    public void close() throws IOException {
+        closeAttributes();
+        super.close();
+    }
+
+    private static final String[] ANSI_COLOR_MAP = {"black", "red",
+            "green", "yellow", "blue", "magenta", "cyan", "white",};
+
+    private static final byte[] BYTES_QUOT = "&quot;".getBytes();
+    private static final byte[] BYTES_AMP = "&amp;".getBytes();
+    private static final byte[] BYTES_LT = "&lt;".getBytes();
+    private static final byte[] BYTES_GT = "&gt;".getBytes();
+
+    public HtmlAnsiOutputStream(OutputStream os) {
+        super(os, // With jdk8 the below could be simplified to:
+                // AnsiConsole.out()::getTerminalWidth
+                new WidthSupplier() {
+                    @Override
+                    public int getTerminalWidth() {
+                        return AnsiConsole.out().getTerminalWidth();
```

Comment:
Using the system terminal width does not make sense.  Try with Integer.MAX_VALUE ?